kytos/core/auth.py

```diff
@@ -136,232 +240,113 @@ def register_core_auth_services(self):
-            if not box:
-                response = {
-                    "answer": f'User with uid {uid} not found',
-                    "code": HTTPStatus.NOT_FOUND.value
-                }
-            elif error:
-                response = {
-                    "answer": "User has not been deleted",
-                    "code": HTTPStatus.INTERNAL_SERVER_ERROR.value,
-                }
-            else:
-                response = {
-                    "answer": "User successfully deleted",
-                    "code": HTTPStatus.OK.value,
-                }
-
-        content = {
-            "box_id": uid,
-            "namespace": self.namespace,
-            "callback": _delete_user_callback,
-        }
-        event = KytosEvent(name="kytos.storehouse.delete", content=content)
-        self.controller.buffers.app.put(event)
-        while True:
-            time.sleep(0.1)
-            if response:
-                break
-        return response["answer"], response["code"]
+    def _delete_user(self, username):
+        """Delete a user using MongoDB."""
+        if self.user_controller.delete_user(username) == {}:
+            raise NotFound(f"User {username} not found")
+        return jsonify(f"User {username} deleted succesfully"), 200
 
     @authenticated
-    def _update_user(self, uid):
-        """Update user data using Storehouse."""
-        response = {}
-
-        def _update_user_callback(_event, box, error):
-            nonlocal response
-            if not box:
-                response = {
-                    "answer": f'User with uid {uid} not found',
-                    "code": HTTPStatus.NOT_FOUND.value
-                }
-            elif error:
-                response = {
-                    "answer": "User has not been updated",
-                    "code": HTTPStatus.INTERNAL_SERVER_ERROR.value,
-                }
-            else:
-                response = {
-                    "answer": "User successfully updated",
-                    "code": HTTPStatus.OK.value,
-                }
-
-        req = request.json
+    def _update_user(self, username):
+        """Update user data using MongoDB."""
+        req = self._get_request()
         allowed = ["username", "email", "password"]
-
         data = {}
         for key, value in req.items():
             if key in allowed:
                 data[key] = value
-
-        content = {
-            "namespace": self.namespace,
-            "box_id": uid,
-            "data": data,
-            "callback": _update_user_callback,
-        }
-        event = KytosEvent(name="kytos.storehouse.update", content=content)
-        self.controller.buffers.app.put(event)
-        while True:
-            time.sleep(0.1)
-            if response:
-                break
-        return response["answer"], response["code"]
+        try:
+            updated = self.user_controller.update_user(username, data)
+        except ValidationError as e:
```

Comment:
Same comment as https://github.com/kytos-ng/kytos/pull/295/files#r1001870597, we'll need to handle the exception here. I'll also encourage you to try what would happen if an `username` is updated with an existing username, would it raise `DuplicateKeyError`?

Reply:
Indeed `DuplicateKeyError` was raised. Fixed

Reply:
Cool, thanks for confirming and adapting this @Alopalao